numpyro/distributions/util.py

```diff
@@ -156,6 +156,8 @@ def _binom_inv_cond_fn(val):
 def _binomial_dispatch(key, p, n):
     def dispatch(key, p, n):
         is_le_mid = p <= 0.5
+        #Make sure p=0 is never taken into account as a fix for possible zeros in p.
```

Comment:
could we just simply clip p to tiny? `jnp.clip(p, minval=jnp.finfo(p.dtype).tiny)`

Reply:
This works for me: ```jnp.clip(p, jnp.finfo(jnp.float32).tiny)```

Reply:
this might affect samples with small `p` in x64. How about updating `_binomial_inversion` to
```
geom = jnp.floor(jnp.log1p(-u) / log1_p) + 1
...
log1_p = jnp.log1p(-p)
log1_p = jnp.where(log1_p == 0, -jnp.finfo(log1_p.dtype).tiny, log1_p)
```
The issue seems to come from `log1_p=0` and `jnp.log1p(-u) < 0`, which leads to a negative `geom`

Reply:
Could you also fix the lint issue? I think you need a space `# comment` for the comment.